src/ipc/ipc4/handler.c

```diff
@@ -420,8 +420,9 @@ static int update_dir_to_pipeline_component(uint32_t *ppl_id, uint32_t count)
 				 * since one of them is for playback and the other one
 				 * is for capture
 				 */
-				if (dev_comp_type(icd->cd) == SOF_COMP_HOST ||
-				    dev_comp_type(icd->cd) == SOF_COMP_DAI)
+				if (dev_comp_type(icd->cd) == SOF_COMP_COPIER &&
```

Comment:
@RanderWang we agreed to stop adding new comps as this had kernel impact. Can we code this differently so it's classed like the FILEREAD/WRITE where we can have a big comment saying it's internal only ?
@plbossart fyi.

Reply:
sure.  I will rework it